spec/helpers.lua

```diff
@@ -186,9 +296,167 @@ Status received:
+    args.n = 1
+    return false
+  end
+  return true, {json}
+end
+say:set("assertion.jsonbody.negative", [[
+Expected response body to contain valid json. Got:
+%s
+]])
+say:set("assertion.jsonbody.positive", [[
+Expected response body to not contain valid json. Got:
+%s
+]])
+luassert:register("assertion", "jsonbody", jsonbody,
+                  "assertion.jsonbody.negative",
+                  "assertion.jsonbody.positive")
+
+---
+-- Adds an assertion to look for a named header in a `headers` subtable.
+-- Header name comparison is done case-insensitive.
+-- @return value of the header
+local function res_header(state, args)
+  local header = args[1]
+  local res = args[2] or kong_state.kong_request or kong_state.kong_response
+  if (type(res) ~= "table") or (type(res.headers) ~= "table") then
+    table.insert(args, 1, "<< 'header' assertion input does not contain a 'headers' subtable >>")
+    table.insert(args, 1, header)
+    args.n = 2
+    return false
+  end
+  local value = lookup(res.headers, header)
+  table.insert(args, 1, res.headers)
+  table.insert(args, 1, header)
+  args.n = 2
+  if not value then
+    return false
+  end
+  return true, {value}
+end
+say:set("assertion.res_header.negative", [[
+Expected header: 
+%s
+But it was not found in: 
+%s
+]])
+say:set("assertion.res_header.positive", [[
+Did not expected header: 
+%s
+But it was found in: 
+%s
+]])
+luassert:register("assertion", "header", res_header,
+                  "assertion.res_header.negative",
+                  "assertion.res_header.positive")
+
+---
+-- Adds an assertion to look for a query parameter in a `queryString` subtable.
+-- Parameter name comparison is done case-insensitive.
+-- @return value of the parameter
+local function req_query_param(state, args)
+  local param = args[1]
+  local req = args[2] or kong_state.kong_request
+  if (not req) and kong_state.kong_response then
+    table.insert(args, 1, "<< 'queryparam' assertion only works with a request object >>")
+    table.insert(args, 1, param)
+    args.n = 2
+    return false
+  end
+  if (type(req) ~= "table") or (type(req.queryString) ~= "table") then
+    table.insert(args, 1, "<< 'queryparam' assertion input does not contain a 'queryString' subtable >>")
+    table.insert(args, 1, param)
+    args.n = 2
+    return false
+  end
+  local value = lookup(req.queryString, param)
+  table.insert(args, 1, req.queryString)
+  table.insert(args, 1, param)
+  args.n = 2
+  if not value then
+    return false
+  end
+  return true, {value}
+end
+say:set("assertion.req_query_param.negative", [[
+Expected query parameter: 
+%s
+But it was not found in: 
+%s
+]])
+say:set("assertion.req_query_param.positive", [[
+Did not expected query parameter: 
+%s
+But it was found in: 
+%s
+]])
+luassert:register("assertion", "queryparam", req_query_param,
+                  "assertion.req_query_param.negative",
+                  "assertion.req_query_param.positive")
```

Comment:
I ran into those for the request transformer plugin. But placed them here as they are generic and might be useful for customers building their own plugins.